### PR TITLE
220262 admin action restart create payment plan

### PR DESCRIPTION
**Pull request by pavlo-mk**
[AB#220262](https://unicef.visualstudio.com/ICTD-HCT-MIS/_workitems/edit/220262): Admin: tweak admin functionality to allow trigger the task to prepare payment plan
Screenshot 2024-11-23 at 9 05 42 PM
Screenshot 2024-11-23 at 9 06 42 PM
Screenshot 2024-11-23 at 9 23 54 PM

### CODE REVIEW

File: src/hct_mis_api/apps/payment/celery_tasks.py [hunk 1]
```diff
@@ -390,6 +391,18 @@ def remove_old_payment_plan_payment_list_xlsx(self: Any, past_days: int = 30) ->
...
+            "task_name": "prepare_payment_plan_task",
+            "payment_plan_id": payment_plan_id,
+        }
+    )
+    if cache.get(cache_key):
+        logger.info(f"Task prepare_payment_plan_task with payment_plan_id {payment_plan_id} already running.")
+        return False
+
+    # 10 hours timeout
+    cache.set(cache_key, True, timeout=60 * 60 * 10)
```
Comment (domdinicola) [inline src/hct_mis_api/apps/payment/celery_tasks.py:405]: 10 hours... maybe we can have 2 or 3

File: tests/selenium/conftest.py [hunk 2]
```diff
@@ -560,14 +560,16 @@ def create_super_user(business_area: BusinessArea) -> User:
...
     business_area.countries.add(country)
-    user = UserFactory.create(
-        is_superuser=True,
-        is_staff=True,
-        username="superuser",
-        password="testtest2",
-        email="test@example.com",
-        partner=partner,
-    )
+    if not (user := User.objects.filter(pk="4196c2c5-c2dd-48d2-887f-3a9d39e78916").first()):
```
Comment (domdinicola) [inline tests/selenium/conftest.py:563]: 🌵  what is this!??!

**Review by johniak — APPROVED** [review]

File: tests/unit/apps/utils/test_payment_plan_celery_tasks_mixin.py [hunk 3]
```diff
@@ -26,14 +33,20 @@ def setUpTestData(cls) -> None:
         cls.user.is_active = True
         cls.user.save()
 
-        cls.payment_plan = PaymentPlanFactory()
+        cls.payment_plan = PaymentPlanFactory(program=cls.program)
 
     def setUp(self) -> None:
         self.url = reverse("admin:payment_paymentplan_change", args=[self.payment_plan.id])
 
+        self.patcher = patch("django.conf.settings.ROOT_TOKEN", "test-token123")
```
Comment (domdinicola) [inline tests/unit/apps/utils/test_payment_plan_celery_tasks_mixin.py:41]: you can use override settings @pavlo-mk

File: tests/selenium/conftest.py (hunk 2 above)
Comment (domdinicola) [reply tests/selenium/conftest.py:563]: @pavlo-mk

Comment (pavlo-mk) [reply tests/selenium/conftest.py:563]: in fixtures we have assigned  user with with this id `4196c2c5-c2dd-48d2-887f-3a9d39e78916` for rid grievances and a lot other models like created_by, imported_by. We need to have the user with this id

Comment (domdinicola) [reply tests/selenium/conftest.py:563]: can't we have a get or create?

Comment (pavlo-mk) [reply tests/selenium/conftest.py]: tried didn't work properly 
https://github.com/unicef/hope/pull/4477/commits/540bf6aed8ded226c5f2038cbab350f24ed3921a

**Review by MarekBiczysko — APPROVED** [review]

**Pull request merged**